src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionsController.kt

```diff
@@ -35,40 +42,67 @@ class IdentificationSessionsController(
     }
 
     @PostMapping
-    fun createSession(@RequestBody createIdentitySessionRequest: CreateIdentitySessionRequest): Mono<ResponseEntity<CreateIdentitySessionResponse>> {
-        // Currently, returns a mock session response
-        return tcTokenUrlService.getTcTokenUrl()
-            .flatMap { tcTokenUrl ->
-                identificationSessionService.save(
-                    tcTokenUrl,
-                    createIdentitySessionRequest.refreshAddress,
-                    createIdentitySessionRequest.requestAttributes
-                )
-            }.map {
+    fun createSession(
+        @RequestBody createIdentitySessionRequest: CreateIdentitySessionRequest,
+        serverHttpRequest: ServerHttpRequest
+    ): Mono<ResponseEntity<CreateIdentitySessionResponse>> {
+        return identificationSessionService
+            .save(
+                refreshAddress = "https://localhost:8443/", // Currently a mock value, later will inject from env var mapped by API Key
+                requestAttributes = createIdentitySessionRequest.requestAttributes
+            ) // PublishOn is suggested by IntelliJ, maybe there is another way to resolve blocking URLEncoder.encode()
+            .publishOn(Schedulers.boundedElastic())
+            .map {
+                val encodedTcTokenUrl = URLEncoder.encode("${serverHttpRequest.uri}/$TCTOKEN_ENDPOINT/${it.useIDSessionId}", Utils.ENCODING)
                 ResponseEntity
                     .status(HttpStatus.OK)
                     .contentType(MediaType.APPLICATION_JSON)
-                    .body(
-                        CreateIdentitySessionResponse(
-                            tcTokenUrl = it.tcTokenUrl,
-                            sessionId = it.sessionId
-                        )
-                    )
-            }.doOnError { exception ->
-                log.error { "error occurred when creating identification session: ${exception.message}" }
-            }.onErrorReturn(
+                    .body(CreateIdentitySessionResponse(tcTokenUrl = encodedTcTokenUrl))
+            }
+            .doOnError { exception ->
+                log.error {
+                    "error occurred when creating identification session: ${exception.message}"
+                }
+            }
+            .onErrorReturn(
+                ResponseEntity.internalServerError().body(null)
+            )
+    }
+
+    @GetMapping("/$TCTOKEN_ENDPOINT/{useIDSessionId}")
```

Comment:
That's not the REST way to do it. It's always `/resource/{resource-id}/resource` so it should be `/sessions/{sessionId}/tc-token`

Reply:
Ah and please add the `produces` parameter of the `GetMapping` annotation to indicate that XML is returned.